projects/train/train/callbacks.py

```diff
@@ -176,17 +159,3 @@ def on_train_epoch_end(self, trainer, pl_module) -> None:
         ckpt_path = os.path.join(tmpdir, "model.pt")
         with open(ckpt_path, "wb") as f:
             torch.jit.save(trace, f)
-
-        # save lightning checkpoint to local
-        ckpt_path = os.path.join(tmpdir, "checkpoint.ckpt")
-        trainer.save_checkpoint(ckpt_path, weights_only=False)
-
-        # Report to train session
-        checkpoint = train.Checkpoint.from_directory(tmpdir)
-        report_with_retries(metrics, checkpoint)
-
-        # Add a barrier to ensure all workers finished reporting here
-        torch.distributed.barrier()
-
-        if self.local_rank == 0:
-            shutil.rmtree(tmpdir)
```

Comment:
Or is there some other equivalent thing that happens?

Reply:
So I made `lightray` automatically add a [custom](https://github.com/EthanMarx/lightray/blob/main/lightray/callbacks.py#L28) `RayTrainReportCallback` that just performs the reporting, but with retries in case of s3 errors. 

And then I refactored out the model tracing to its own callback

Reply:
The Idea being all tune jobs will want to do the reporting, so makes sense to always have that functionalty